01-JS-DOM-APIs/main.js

```diff
@@ -1,31 +1,58 @@
 
-//when the window is loaded "hello world" appears with a animation.
+//when the window is loaded "hello world" appears with a animation(Fade in).
+
 window.addEventListener("load",() => {
     let hdn = document.getElementsByClassName("hidden-text")[0];
     hdn.innerHTML = "Hello World!";
 });
 
+ // ************************************************************************
+ // ************************************************************************
+ // ************************************************************************
 
-let btn =  document.getElementsByClassName("joke")[0];
-
-//add a event to the button
-btn.addEventListener("click",get_joke);
 
+//reusable_fun()
+//take:(string,string,bool) return: Promise
+//Gets the information from the taked API via XMLhttprequest
+function reusable_fun (mtd, url, asyn) {
+    let myPromise = new Promise( (resolve,reject) => {
+        let xhttp = new XMLHttpRequest();
+        xhttp.open(mtd, url, asyn);
+        xhttp.onreadystatechange = () => {
+            if(xhttp.readyState == 4 && xhttp.status == 200){
+                resolve(xhttp.responseText); 
+            } else {
+                reject;
```

Comment:
The reject method is missing the closing parenthesis, it should be reject(). If you add that, you will probably encounter an error because another if is needed. add an else if (condition when errors occurs).